### PR TITLE
Define error values for missing x5c header and unsupported elliptic curves

### PR DESCRIPTION
**Pull request by milosgajdos**
Most of the errors in the shared source have a dedicated error value
except for a couple that got away:
* x5c header missing in the JWT header
* An unsupported/unknown elliptic curve is found

This commit defines the error values for those cases.

This would be helpful to https://github.com/distribution/distribution project when discriminating between different errors during token validation.

### CODE REVIEW

**Comment by milosgajdos** [comment]
@mcpherrinm @mitar you folks mind having a look at this? It would help us with the next release of distribution (aka registry). Sorry for bugging you like this - as a maintainer of a large project I know how annoying these pings are 😇

**Review by mcpherrinm — APPROVED** [review]

**Comment by mcpherrinm** [comment]
No problem. This looks fine by me, but I’m on vacation for a few days so will get somebody else to take a second look and merge. Do you need a tagged version after this merges?

**Review by mitar — APPROVED** [review]
Looks good to me as well.

**Review by aarongable — APPROVED** [review]

**Comment by aarongable** [comment]
Merged and tagged: https://github.com/go-jose/go-jose/releases/tag/v4.0.2

**Pull request merged**
